File: frontend/src/types/job.ts
export type JobStatus =
  | "saved"
  | "applied"
  | "networking"
  | "interviewing"
  | "offer"
  | "rejected";

export interface Job {
  id: string;
  company_name: string;
  role_title: string;
  posting_url: string | null;
  location_remote_status: string | null;
  jd_text: string | null;
  status: JobStatus;
  date_applied: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface JobCreate {
  company_name: string;
  role_title: string;
  posting_url?: string;
  location_remote_status?: string;
  jd_text?: string;
  status: JobStatus;
  date_applied?: string;
  notes?: string;
}

export interface JobUpdate {
  company_name?: string;
  role_title?: string;
  posting_url?: string;
  location_remote_status?: string;
  jd_text?: string;
  status?: JobStatus;
  date_applied?: string;
  notes?: string;
}

export interface JobDetail extends Job {
  contacts: import("@/types/contact").ContactWithChats[];
  interviews: import("@/types/interview").Interview[];
}

File: backend/app/models/job.py
import uuid
import enum
from datetime import datetime

from sqlalchemy import Column, String, Text, Date, DateTime, Enum, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from app.database import Base


class JobStatus(str, enum.Enum):
    saved = "saved"
    applied = "applied"
    networking = "networking"
    interviewing = "interviewing"
    offer = "offer"
    rejected = "rejected"


class Job(Base):
    __tablename__ = "jobs"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=True, index=True)
    company_name = Column(String, nullable=False)
    role_title = Column(String, nullable=False)
    posting_url = Column(String, nullable=True)
    location_remote_status = Column(String, nullable=True)
    jd_text = Column(Text, nullable=True)
    status = Column(Enum(JobStatus), nullable=False, default=JobStatus.saved)
    date_applied = Column(Date, nullable=True)
    notes = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
        nullable=False,
    )

    contacts = relationship("Contact", back_populates="job", cascade="all, delete-orphan")
    interviews = relationship("Interview", back_populates="job", cascade="all, delete-orphan")
